features/FEATURE_PSA/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,600 @@
+
+If an Armv8-M PSA target cannot inherit from `PSA_V8_M` because it has to
+inherit from one of it's family targets, then following attributes have to be
+added.
+```json
+        "features_add": [
+            "PSA"
+        ],
+        "extra_labels_add": [
+            "TFM",
+            "TFM_V8M"
+        ],
+        "macros_add": [
+            "MBEDTLS_PSA_HAS_ITS_IO",
+            "MBEDTLS_USE_PSA_CRYPTO"
+        ]
+```
+
+#  Enabling PSA at application level
+Having an entropy source is crucial for Mbed TLS and Mbed Crypto. The
+[document](https://os.mbed.com/docs/mbed-os/latest/porting/entropy-sources.html)
+talks about entropy and how to add an entropy source. Sometimes a target might
+not have a True Random Number Generator (TRNG), in that case the target will be
+configured as a non-PSA target in `targets/targets.json`. In that scenario, if
+an application wants to use that target as a PAS target then it is the
+responsibility of application to provide an entropy source and mark that target
+as PSA target at application level. The config option
+[target_overrides](https://os.mbed.com/docs/mbed-os/latest/reference/configuration.html)
+can be used to enable PSA for a target.
+
+example mbed_app.json:
+```json
+"target_overrides": {
+    "K64F": {
+        "inherits": ["PSA_V7_M"]
+    }
+}
+```
+
+# Build and validation
+For dual-core and Armv8-M PSA targets, TF-M runs on the SPE and provides PSA
+services. The python script `build_tfm.py` automates building TF-M and copying
+the TF-M binary to a predefined location defined by the target attribute
+`tfm_delivery_dir`. Another python script `build_psa_compliance.py` automates
+building PSA API compliance tests.
+
+The
+[mbed-os-tf-m-regression-tests](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests)
+contains build scripts, TF-M regression tests and PSA API compliance tests.
+
+## Building TF-M and running regression tests
+Follow the steps below to build TF-M and regression tests:
+
+1. Clone
+   [mbed-os-tf-m-regression-tests](https://github.com/ARMmbed/mbed-os-tf-m-regression-tests)
+   repo
+1. Update `mbed-os.lib` to the version of Mbed OS that contains new target
+   support including `targets.json` changes described in this document
+1. Run `mbed deploy`
+1. Run `python3 build_tfm.py -m <new target> -t <toolchain> -c
+   ConfigRegressionIPC.cmake`
+    * The command builds necessary services in TF-M to run regression tests
+    *NOTE*\
+    Use this config **ONLY TO** run regression tests. The secure binary copied
+    to Mbed OS **MUST** be generated using the config option
+    `ConfigCoreIPC.cmake`.
+1. Ensure TF-M binary is copied to location defined by target attribute
+   `tfm_delivery_dir`
+1. Run `mbed compile -m <new target> -t <toolchain>` to build Mbed OS and run
+   regression tests
+1. Flash the regression tests binary on to the target and ensure all regression
+   tests passes
+
+*TIP*\
+Supported options:
+```console
+usage: build_tfm.py [-h]
```

Comment:
It would be good to state `build_tfm.py -h` to see available options that avoids any updates to this section in the future if there any stales.